scripts/preprocess_data.py

```diff
@@ -108,70 +179,101 @@ def main():
     if args.shuffle:
         random.shuffle(collections.train)
 
-    with h5py.File(args.h5_prefix + "train.h5", "w") as f:
-        # split collections.train into batches and save them to hdf5
-        split_train, drop_last = split_array(collections.train, args.batch_size)
-        f.attrs["drop_last"] = drop_last
-        for i, batch in enumerate(split_train):
-            save_configurations_as_HDF5(batch, i, f)
+    # split collections.train into batches and save them to hdf5
+    split_train = np.array_split(collections.train,int(os.cpu_count()/4))
+    drop_last = False
+    if len(collections.train) % 2 == 1:
```

Comment:
Why do we only drop last when the length of the initial collection is an odd number? I would think that we need to check if `(len(collections.train) // num_process) % batch_size != 0` or `((len(collections.train) // num_process) +1 ) % batch_size != 0` (The latter in the case in which `collections.train` is not evenly divisible into `num_process` sub-arrays)

Reply:
The way we have it setup the batch_size doesn't matter. We are just separating the data based on cpu count and checking the drop_last is based on the odd number in split_array.

Reply:
Correct me if i'm wrong, but I think `drop_last` is not primarily used in the chunking of the data, rather it's an argument to the dataloader which tells the pytorch dataloader whether to always return uniform batch sizes (and therefore drop the last remainders)?

Reply:
I am not sure, but the logic to set drop last was based off of the input being divisible by 2 in the original. I took that and forewent the rest of split_array.

Reply:
def split_array(a: np.ndarray, max_size: int):
    drop_last = False
    if len(a) % 2 == 1:
        a = np.append(a, a[-1])
        drop_last = True